src/components/notes/sharing.jsx

```diff
@@ -0,0 +1,67 @@
+import React, { useCallback, useState, useMemo, useEffect } from 'react'
+import SharingProvider, { ShareButton, ShareModal } from 'cozy-sharing'
+import { withClient } from 'cozy-client'
+import withLocales from 'cozy-sharing/dist/withLocales'
+
+const LocalizedSharingProvider = withLocales(SharingProvider)
+
+function normalizeFile(id, file, parent) {
+  const type = (file && (file.type || file._type)) || 'io.cozy.files'
+  const path = parent ? parent.path + '/' + file.name : undefined
+  return {
+    _id: id,
+    id: id,
+    _type: type,
+    type: type,
+    path: path,
+    ...(file || {})
+  }
+}
+
+export default withClient(function SharingWidget(props) {
+  const { client } = props
+  const [parent, setParent] = useState(undefined)
+
+  const id = props.fileId || props.file.id || props.file._id
+
+  const file = useMemo(
+    () => {
+      return normalizeFile(id, props.file, parent)
+    },
+    [id, parent]
+  )
+
+  useEffect(
+    () => {
+      async function getParent(file) {
+        const parent = await client.query(
+          client.get('io.cozy.files', file.attributes.dir_id)
+        )
```

Comment:
Not sure about that.
An exception would throw the effect but will have no other consequences. The not catched exception would then appear in the console log. 
What whould you want to do better if you catch the error ?

Reply:
Sending it to sentry for instance or Detect if the error is “no network” and then do something specific with this information ? 

Also if we start to use errorboundary (we should start using it tbh. We already use it on banks) it will rise the exception. we should think about the behavior of our app when errors occur.

Reply:
done